include/mpm_explicit.tcc

```diff
@@ -444,6 +452,40 @@ bool mpm::MPMExplicit<Tdim>::apply_nodal_tractions() {
   return status;
 }
 
+//! Assign materials to particle sets
```

Comment:
Previously we defined `assign_material` in mesh, which makes sense for distributed parallelisation. Moving it here causes problems. We need to discuss this.